File: OneDrive/Desktop/COEXIST/backcoexist-final1SpringBoot/src/main/java/tn/esprit/coexist/service/BookingService.java
package tn.esprit.coexist.service;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import tn.esprit.coexist.entity.Booking;
import tn.esprit.coexist.entity.Carpooling;
import tn.esprit.coexist.entity.FeedBack;

import java.util.List;
import java.util.Optional;


public interface BookingService {
  public Booking addBooking(Booking booking, Integer carpoolingID,Integer userId);
  public void deleteBooking(Integer bookingId);
  public List<Booking> getALLBooking();
  public void sendDelateCarpoolingEmail(String email, Integer carpoolingID);
  public void sendDelateCarpoolingEmail();
  public Optional<Booking> getBookingById(Integer bookingID);
  public Carpooling getCarpoolingForBooking(Integer bookingID);

}

File: OneDrive/Desktop/COEXIST/FrontCoexist-final1Angular/FrontCoexist-final1/src/app/Services/booking.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Carpooling} from "../entity/Carpooling";
import {Observable} from "rxjs";
import {Booking} from "../entity/Booking";
import * as Twilio from 'twilio';

@Injectable({
  providedIn: 'root'
})
export class BookingService {


  private baseUrl = 'http://localhost:8000/';

  constructor(private http: HttpClient) { }
  addBooking(booking: Booking, carpoolingID: number, userId: number): Observable<Booking> {
    const url = `${this.baseUrl}addBooking/${carpoolingID}/${userId}`;
    return this.http.post<Booking>(url, booking);
  }
  getAllBookings(): Observable<Booking[]> {
    return this.http.get<Booking[]>(this.baseUrl + 'getALLBooking');
  }


  deleteBooking(bookingId: number): Observable<void> {
    const url = `${this.baseUrl}deleteBooking/${bookingId}`;
    return this.http.delete<void>(url);
  }

  getBookingByBookingID(bookingID: number): Observable<Booking> {
    const url = `${this.baseUrl}${bookingID}`;
    return this.http.get<Booking>(url);
  }
  getCarpoolingForBooking(bookingID: number): Observable<Carpooling> {
    const url = `${this.baseUrl}carpooling/${bookingID}`;
    return this.http.get<Carpooling>(url);
  }
}

